src/ray/gcs/gcs_server/gcs_resource_manager.cc

```diff
@@ -111,18 +112,21 @@ void GcsResourceManager::HandleGetAllAvailableResources(
 
 void GcsResourceManager::UpdateFromResourceReport(const rpc::ResourcesData &data) {
   NodeID node_id = NodeID::FromBinary(data.node_id());
-  if (RayConfig::instance().gcs_actor_scheduling_enabled()) {
-    UpdateNodeNormalTaskResources(node_id, data);
-  } else {
-    if (!cluster_resource_manager_.UpdateNodeAvailableResourcesIfExist(
-            scheduling::NodeID(node_id.Binary()), data)) {
-      RAY_LOG(INFO)
-          << "[UpdateFromResourceReport]: received resource usage from unknown node id "
-          << node_id;
+  // Only need to update worker nodes' resource usage.
+  if (node_id != local_node_id_) {
```

Comment:
```
if (node_id == local_node_id_) {
    return;
}

if (RayConfig::instance().gcs_actor_scheduling_enabled()) {
  UpdateNodeNormalTaskResources(node_id, data);
} else {
  if (!cluster_resource_manager_.UpdateNodeAvailableResourcesIfExist(
              scheduling::NodeID(node_id.Binary()), data)) {
        RAY_LOG(INFO)
            << "[UpdateFromResourceReport]: received resource usage from unknown node id "
            << node_id;
  }
}
UpdateNodeResourceUsage(node_id, data);
```

Reply:
>   // Only need to update worker nodes' resource usage.
  if (node_id != local_node_id_) {

Hmm I don't understand this part. Why do we only need to update worker nodes' resource usage?

Reply:
We don't run any normal tasks or actors in gcs node (server). Even with gcs scheduler enabled, gcs server only queues and schedules tasks, without allocating any **local** resources to them. In terms of scheduling, gcs node actually has **zero** total resources, so we don't need to update it here.